src/hooks/use-sorting/use-sorting.js

```diff
@@ -15,14 +15,29 @@ const sortItems = (items, field, direction, sortingFunction = sortBy) => {
   return sortedItems;
 };
 
+const getInitialState = (items, field, sortDirection, sortingFunction) => ({
+  items: sortItems(items, field, sortDirection, sortingFunction),
+  sortedBy: field,
+  sortDirection,
+});
+
+const useSortingState = (items, field, sortDirection, sortingFunction) => {
+  const [sortState, setSorting] = React.useState(
+    getInitialState(items, field, sortDirection, sortingFunction)
```

Comment:
This should return a function maybe? So `getInitialSortState = ({...}) => () => ({...})`. Then we avoid the sorting inside the func on each render.

Reply:
Ah, got it! It's the [lazy state initialization](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state) thing 😄 
Thanks!

Reply:
Yup, exactly that. Sorry for not calling the beauty by name earlier.